types/coin_test.go

```diff
@@ -421,20 +453,29 @@ func TestCoins(t *testing.T) {
 		{"tree", NewInt(0)},
 		{"mineral", NewInt(1)},
 	}
-	dup := Coins{
+	dup1 := Coins{
 		{"gas", NewInt(1)},
 		{"gas", NewInt(1)},
 		{"mineral", NewInt(1)},
 	}
+	notDup := Coins{
+		{"GAS", NewInt(1)},
+		{"gAs", NewInt(1)},
+	}
 	neg := Coins{
 		{"gas", NewInt(-1)},
 		{"mineral", NewInt(1)},
 	}
 
```

Comment:
add a test where first denom is valid and rest are invalid (or one invalid later on)